include/rocksdb/options.h

```diff
@@ -1946,6 +1946,23 @@ struct ReadOptions {
   // Default: false
   bool allow_unprepared_value = false;
 
+  // EXPERIMENTAL
+  //
+  // Long-running iterators are holding onto memory and storage resources long
+  // after they are obsolete. This setting (when enabled) will fix that problem
+  // for as long as iterator periodically makes some progress and its supplied
+  // `read_options` was configured with non-nullptr `snapshot` value.
+  // The feature is engineered so that the performance impact should be
+  // negligible. We expect the default value to be true some time in the future.
+  //
+  // NOTE 1: Does not have effect on TransactionDB with WRITE_PREPARED or
+  //         WRITE_UNPREPARED policies.
+  //
+  // NOTE 2: Not recommended if your application is reading UDT timestamps.
```

Comment:
I think we can be more specific. "True is not recommended if using user-defined timestamp with persist_user_defined_timestamps=false and non-nullptr ReadOptions::timestamp or ReadOptions::iter_start_ts, because an auto-refreshing iterator will not prevent user timestamp information from being dropped during iteration. Auto-refresh might be disabled for this combination in the future."